types/mytime.go

```diff
@@ -337,30 +337,34 @@ func calcWeek(t *MysqlTime, wb weekBehaviour) (year int, week int) {
 
 // mixDateAndTime mixes a date value and a time value.
 func mixDateAndTime(date, time *MysqlTime, neg bool) {
-	if !neg && time.hour < 24 {
```

Comment:
Is it safe to remove this check?

Reply:
Actually, these lines are useless, they are used to avoid the calculation outside of this check.
After this commit, `time.hour < 24` is promised to be true, so this check should be removed.

Reply:
> time.hour < 24 is promised to be true

How?